src/cr/cube/cube.py

```diff
@@ -784,20 +796,21 @@ class _UnweightedCountMeasure(_BaseMeasure):
 
     @lazyproperty
     def _flat_values(self):
-        """tuple of int counts before weighting."""
-        if (
-            self._cube_dict["result"]["measures"]
-            .get("valid_count_unweighted", {})
-            .get("data")
-        ):
-            # ---If valid_count are expressed in the cube dict, returns its data.
-            # ---This condition can happen in case of numeric array cube response.
-            # ---Under this circumstances the numeric array measures will contain the
-            # ---mean measure and a valid count measure for the unweighted counts.
-            return tuple(
-                self._cube_dict["result"]["measures"]["valid_count_unweighted"]["data"]
-            )
-        return tuple(self._cube_dict["result"]["counts"])
+        """np.ndarray of np.float64 counts before weighting or None, if unavailable.
+
+        Use floats to avoid int overflow bugs and so we can use nan.
+        """
```

Comment:
One convention I've adopted, which is consistent with Python 3 type hints, is to prefix are return value that can also be None as "Optional ". So this would become "Optional 1D np.float64 ndarray of counts before weighting."

Then in the body of the docstring you can mention the circumstances under which it is None, like "This value is `None` when the cube-result does not contain a cube-counts measure."

Reply:
Hm, but this was mislabelled because it never returns None, right?